### PR TITLE
Enable specifying output assembly name in compiler options

### PR DESCRIPTION
**Pull request by moozzyk**
Addresses 1797

### CODE REVIEW

**Comment by moozzyk** [comment]
#1797

File: test/dotnet-compile.Tests/CompilerTests.cs [hunk 1]
```diff
@@ -167,6 +167,36 @@ public void DepsDependencyContextIsValidOnBuild()
                 runCommand.Execute().Should().Pass();
         }
 
+        [Fact]
+        public void CanSetOutputAssemblyName()
+        {
+            // create unique directories in the 'temp' folder
+            var root = Temp.CreateDirectory();
```
Comment (davidfowl) [inline test/dotnet-compile.Tests/CompilerTests.cs]: His should be using the test framework

Comment (moozzyk) [inline test/dotnet-compile.Tests/CompilerTests.cs]: Is it this `TestAssetsManager` thing?

Comment (davidfowl) [inline test/dotnet-compile.Tests/CompilerTests.cs]: Yea

**Comment by moozzyk** [comment]
:up::date:

**Comment by moozzyk** [comment]
I think some changes will be needed to support this in P2P scenario - I will do this separately though

**Comment by moozzyk** [comment]
P2P references work with this change since they use `CompilationOutputFiles` class to resolve the target assembly and this class is updated accordingly. So no additional work for P2P is needed.

**Comment by davidfowl** [comment]
Looks good, why is the ci failing?

**Comment by moozzyk** [comment]
OSX is failing during restore due to: error: End of Central Directory record could not be found.
On CentOs one test did not pass but does not seem related to my changes and it did not fail on other OSes.
I will restart the builds and investigate CentOS failure if it persists.

**Comment by moozzyk** [comment]
@piotrpMSFT @livarcocc Can anyone merge this? OSX failures are due to corrupted packages - I pinged folks separately on this.

**Comment by davidfowl** [comment]
why is the CI failing?

**Comment by livarcocc** [comment]
@moozzyk This seems like our old CI. @Sridhar-MS has updated our CI and it seems your PR is out in the limbo. I had the same problem. What I ended up doing was closing this PR. Creating a new branch, porting the commits over and creating a new PR from the new branch.

**Comment by moozzyk** [comment]
Thanks @livarcocc - will try that.

**Comment by moozzyk** [comment]
Closing to reopen in a new branch.

**Pull request closed without merging**